net/instaweb/rewriter/webscale_make_scripts_async.cc

```diff
@@ -41,59 +40,53 @@ void WebscaleMakeScriptsAsync::StartDocumentImpl() {
 
 
 void WebscaleMakeScriptsAsync::StartElementImpl(HtmlElement* element) {
-  message_handler = driver()->message_handler();
+  MessageHandler* message_handler = driver()->message_handler();
 
   // Script element found.
   if (element->keyword() == HtmlName::kScript) {
     const char* src_attribute = element->EscapedAttributeValue(HtmlName::kSrc);
     // Script element has a 'src' attribute.
     if (src_attribute != NULL) {
-      // Convert the src element to a GoogleString format for compatibility.
-      GoogleString* src_string = new GoogleString(src_attribute);
-
       // If there are no custom urls mentioned, print a message and do nothing.
       if (escaped_urls == "") {
-        message_handler->Message(
-          kInfo, "No custom urls provided.");
+        message_handler->Message(kInfo, "No custom urls provided.");
       }
       else {
         // A full match of the custom url needs to be found.
-        bool match = RE2::FullMatch(src_string->c_str(), escaped_urls.c_str());
+        bool match = RE2::FullMatch(src_attribute, escaped_urls.c_str());
         if (match) {
           // If a match is found, add the async attribute.
           driver()->AddAttribute(element, HtmlName::kAsync, "true");
           // Set the debug comment so that it will be displayed when
-          // ModPagespeedEnabledFilters=+debug is used.
+          // ModPagespeedFilters=+debug is used.
           driver()->InsertDebugComment("Webscale added an async attribute successfully", element);
-          message_handler->Message(
-            kInfo, "Adding an async attribute to %s.", src_string->c_str());
+          message_handler->Message(kInfo, "Adding an async attribute to %s.", src_attribute);
         } else {
-            message_handler->Message(
-              kInfo, "Not adding an async attribute to %s.", src_string->c_str());
+            message_handler->Message(kInfo, "Not adding an async attribute to %s.", src_attribute);
         }
       }
     }
   }
 }
 
-// Construct a regular expressions with the custom urls provied. Each custom
+// Construct a regular expression with the custom urls provied. Each custom
 // url will be escaped and an OR of all the escaped urls will be constructed
 // for pattern matching.  Pattern matching is preferred here instead of
-// iterating the list of cusotm urls every time a src attribute is encoutered.
-// This make comparison faster.
+// iterating the list of custom urls every time a src attribute is encountered.
+// This makes comparison faster.
 GoogleString WebscaleMakeScriptsAsync::ConstructPatternFromCustomUrls(const RewriteOptions* options) {
   const int number_of_custom_urls = options->num_custom_async_urls();
+  GoogleString prefix = "";
   GoogleString pattern = "";
 
   if (number_of_custom_urls == 0) {
```

Comment:
Now that you've made the change to the for loop as Augusto suggested, this if() is unnecessary right?